lua/ulx/modules/sh/cfc_checklua.lua

```diff
@@ -2,40 +2,114 @@ CFCUlxCommands.checklua = CFCUlxCommands.checklua or {}
 local cmd = CFCUlxCommands.checklua
 
 local CATEGORY_NAME = "Utility"
+local AUTO_BAN_LENGTH = ( 60 * 60 ) * 6
+local AUTO_BAN_REASON = "Server/Client mismatch. Please disable any third-party programs that modify your game."
+local CONVARS_TO_CHECK = { "sv_allowcslua" }
 
 if SERVER then
-    util.AddNetworkString( "CFC_ULX_StatCheckCL" )
-    util.AddNetworkString( "CFC_ULX_StatCheckSV" )
-end
+    util.AddNetworkString( "CFC_ULX_StatCheck" )
+    util.AddNetworkString( "CFC_ULX_AutoStatCheck" )
+
+    local awaitingResponse = {}
+
+    local function sendAwarn( ply )
+        if not awarn_warnplayer then return end
+
+        awarn_warnplayer( nil, ply, AUTO_BAN_REASON )
+    end
+
+    local function receiveConvars( ply )
+        local any = false
+        local badConvars = {}
+
+        for _, convar in pairs( CONVARS_TO_CHECK ) do
+            local convarValue = net.ReadBool()
+
+            if convarValue then
+                any = true
+                badCovnars[convar] = convarValue
+                ulx.fancyLogAdmin( awaitingResponse[ply], true, "#T's " .. convar .. "value is " .. tostring( convarValue ), ply )
+            end
+        end
+
+        return any, badConvars
+    end
+
+    local function unawaitTimerName( steamId )
+        return "CFC_ULX_AutoStatsUnAwait_" .. steamId
+    end
+
+    net.Receive( "CFC_ULX_StatCheck", function( _, ply )
+        if not awaitingResponse[ply] then return end
+        awaitingResponse[ply] = nil
+
+        receiveConvars( ply )
+    end )
 
-local awaitingResponse = {}
+    net.Receive( "CFC_ULX_AutoStatCheck", function( _, ply )
+        if not awaitingResponse[ply] then return end
+        awaitingResponse[ply] = nil
+        timer.Remove( unawaitTimerName( ply:GetSteamID() ) )
 
-net.Receive( "CFC_ULX_StatCheckSV", function( _, ply )
-    if not awaitingResponse[ply] then return end
-    local convar = net.ReadBool()
-    ulx.fancyLogAdmin( awaitingResponse[ply], true, "#T's sv_allowcslua value is " .. tostring( convar ), ply )
+        local anyConvars, badConvars = receiveConvars( ply )
 
-    awaitingResponse[ply] = nil
-end )
+        if anyConvars then
+            print( "Banning " .. tostring( ply ) .. " for having bad convars." )
+            PrintTable( badConvars )
 
-function cmd.checkluaPlayers( callingPlayer, targetPlayers )
-    for _, ply in pairs( targetPlayers ) do
-        awaitingResponse[ply] = callingPlayer
-        net.Start( "CFC_ULX_StatCheckCL" )
-        net.Send( ply )
+            ulx.ban( nil, ply, AUTO_BAN_LENGTH, AUTO_BAN_REASON )
```

Comment:
If banning kicks, you might wanna warn first or something?
Worried ply will be null after this call

Reply:
ooh ya good call